assignment1/kthInLinkedList_test.py

```diff
@@ -0,0 +1,26 @@
+"""Tests for kthInLinkedList()."""
+
+from kthInLinkedList import *
+import unittest
+
+class kthInLinkedListTest(unittest.TestCase):
+    def setUp(self):
+        self.n = 15
+        self.head = Node(self.n - 1)
```

Comment:
That's correct.

In Python there is no modifier to specify a function or method is private.

The way this is achieve is by adding one or more `_` in front of the name.

For a module function, adding `_` in front of the name, e.g.:

```python
def _SortByDate(values):
  """Sorts the string values as if they were dates in the form DD/MM/YYYY.

  Args:
    values: a list of strings, each string representing a date.

  Returns:
    a list of strings, sorted in date order.
  """
  [...]
```
makes the function private, and, while it can be used as part of the implementation of the module, it should not be used outside that context.

For class member functions and variables, the rule are a little bit different.

A single `_` is used to identify members that are meant to be used by sub-classes but no other code outside the class itself. A double `_` is used to identify members that are meant to be private, so used neither by sub-classes nor by other code outside the class itself.

There are also some special member functions, like `__init__` to initialize an object or `__str__` to implement the to-string conversion, that are not considered private member functions; the names of these both start and end with `__`.

In this case, you probably want to define a private function at the top-level in the module named `_createReverseSequenceOfLength`.